# mandatory arg
string arg

# optional argopt
string[] argopt

# arglist
string[] arglist